Sources/TuistKit/Commands/GenerateCommand.swift

```diff
@@ -19,24 +19,52 @@ struct GenerateCommand: ParsableCommand, HasTrackableParameters {
     )
     var path: String?
 
+    @Argument(help: """
+    A list of targets in which you'd like to focus. \
+    Those and their dependant targets will be generated as sources. \
+    If no target is specified, the project defined targets will be focused.
+    """)
+    var sources: [String] = []
+
+    @Flag(
+        name: .shortAndLong,
+        help: "Don't open the project after generating it."
+    )
+    var noOpen: Bool = false
```

Comment:
Id keep this as `open` and default to not opening as it's the behavior you want in most contexts.

Reply:
Why do you think the behavior is mostly to not open? Which scenario you have in mind?

My scenarios are:
- locally -> 99% I want to open it
- CI -> it's not a problem to pass `--no-open` to avoid opening

Reply:
So i'm thinking about the use case where you generate the project originally (want open for this). For subsequent runs though for example if i add a module, i just usually generate while Xcode is opened so i don't need it to open again. 

Although i'm not sure opening would cause issues there, we could also default to not opening on CI and address most of the common cases by default?

Reply:
Is there analytics data on how frequently this option is used (for both generate/focus)?

Reply:
Makes sense, in our case Xcode is not able to pick up the new project without reopening (probably because we have dependencies and hence the Xcode workspay?), I wonder what is the common case (I would lean toward having dependencies)

For the CI I usually prefer to avoid magic, you will just need to pass --no-open when configuring the script